jgrapht-core/src/main/java/org/jgrapht/graph/SpecificsGraphIterables.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * (C) Copyright 2020-2020, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * See the CONTRIBUTORS.md file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the
+ * GNU Lesser General Public License v2.1 or later
+ * which is available at
+ * http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR LGPL-2.1-or-later
+ */
+package org.jgrapht.graph;
+
+import java.util.Objects;
+
+import org.jgrapht.Graph;
+import org.jgrapht.GraphIterables;
+import org.jgrapht.graph.specifics.Specifics;
+
+/**
+ * An implementation of the {@link GraphIterables} interface using the graph specifics.
+ * 
+ * <p>If the specifics support iterables, then this implementation can employ them. Otherwise
+ * we inherit the default implementation. 
+ * 
+ * @author Dimitrios Michail
+ *
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ */
+public class SpecificsGraphIterables<V, E>
```

Comment:
Is there a reason for introducing this before we have a use-case for it?

Reply:
The `getAllEdges()` can be implemented using this one.

Reply:
Removed it until it is really needed.